File: src/main/java/org/jukeboxmc/command/internal/EnchantCommand.java
package org.jukeboxmc.command.internal;

import com.nukkitx.protocol.bedrock.data.command.CommandParamType;
import org.apache.commons.lang3.EnumUtils;
import org.apache.commons.lang3.math.NumberUtils;
import org.jukeboxmc.Server;
import org.jukeboxmc.command.Command;
import org.jukeboxmc.command.CommandData;
import org.jukeboxmc.command.CommandParameter;
import org.jukeboxmc.command.CommandSender;
import org.jukeboxmc.command.annotation.Description;
import org.jukeboxmc.command.annotation.Name;
import org.jukeboxmc.command.annotation.Permission;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.item.enchantment.Enchantment;
import org.jukeboxmc.item.enchantment.EnchantmentType;
import org.jukeboxmc.player.Player;

import java.util.stream.Collectors;
import java.util.stream.Stream;

/**
 * @author LucGamesYT
 * @version 1.0
 */
@Name("enchant")
@Description("Adds an enchantment to a player's selected item.")
@Permission("jukeboxmc.command.enchant")
public class EnchantCommand extends Command {

    public EnchantCommand() {
        super( CommandData.builder()
                .setParameters( new CommandParameter[]{
                        new CommandParameter( "player", CommandParamType.TARGET, false ),
                        new CommandParameter( "enchantment", Stream.of( EnchantmentType.values() ).map( enchantmentType -> enchantmentType.name().toLowerCase() ).collect( Collectors.toList() ), false ),
                        new CommandParameter( "level", CommandParamType.INT, true ),
                } )
                .build() );
    }

    @Override
    public void execute( CommandSender commandSender, String command, String[] args ) {
        if ( commandSender instanceof Player player ) {
            if ( args.length >= 2 ) {
                Player target = Server.getInstance().getPlayer( args[0] );
                if ( target == null ) {
                    player.sendMessage( "§cThe player " + args[0] + " could not be found" );
                    return;
                }
                String enchantmentValue = args[1].toLowerCase();
                if ( !EnumUtils.isValidEnum( EnchantmentType.class, enchantmentValue.toUpperCase() ) ) {
                    commandSender.sendMessage( "§cEnchantment not found." );
                    return;
                }
                EnchantmentType enchantmentType = EnchantmentType.valueOf( enchantmentValue.toUpperCase() );

                int level = 1;
                if ( args.length >= 3 ) {
                    if ( !NumberUtils.isCreatable( args[2] ) ) {
                        commandSender.sendMessage( "§cThe level must be a number." );
                        return;
                    }
                    level = Integer.parseInt( args[2] );
                }
                Item itemInHand = player.getInventory().getItemInHand();
                if ( itemInHand.getType().equals( ItemType.AIR ) ) {
                    player.sendMessage( "§cAir can not be enchanted." );
                    return;
                }
                Enchantment enchantment = itemInHand.getEnchantment( enchantmentType );
                if ( enchantment != null ) {
                    player.sendMessage( "§cThe item already has the enchantment" );
                    return;
                }
                itemInHand.addEnchantment( enchantmentType, level );
                player.getInventory().setItemInHand( itemInHand );
                player.sendMessage( "Enchanting succeeded for " + target.getName() );
            }
        } else {
            commandSender.sendMessage( "§cYou must be a player." );
        }
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/command/internal/EnchantCommand.kt
package org.jukeboxmc.server.command.internal

import org.apache.commons.lang3.EnumUtils
import org.apache.commons.lang3.math.NumberUtils
import org.jukeboxmc.api.command.Command
import org.jukeboxmc.api.command.CommandSender
import org.jukeboxmc.api.command.ParameterType
import org.jukeboxmc.api.command.annotation.*
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.api.item.enchantment.EnchantmentType
import org.jukeboxmc.api.player.Player
import org.jukeboxmc.server.JukeboxServer
import java.util.*

@Name("enchant")
@Description("Adds an enchantment to a player's selected item.")
@Permission("jukeboxmc.command.enchant")
@Parameters(
    parameter = [
        Parameter("player", ParameterType.TARGET, optional = false),
        Parameter(
            "enchantment", enumValues = arrayOf(
                "aqua_affinity", "bane_of_arthropods",
                "blast_protection", "channeling", "curse_of_binding", "curse_of_vanishing",
                "depth_strider", "efficiency", "feather_falling", "fire_aspect", "fire_protection",
                "flame", "fortune", "impaling", "infinity", "knockback", "looting", "loyalty",
                "luck_of_the_sea", "lure", "mending", "multishot", "piercing", "power", "projectile_protection",
                "protection", "punch", "quick_charge", "respiration", "riptide", "sharpness", "silk_touch",
                "smite", "soul_speed", "thorns", "unbreaking", "swift_sneak"
            ), optional = false
        ),
        Parameter("level", ParameterType.INT, optional = false)
    ]
)
class EnchantCommand : Command {

    override fun execute(commandSender: CommandSender, command: String, args: Array<String>) {
        if (commandSender is Player) {
            if (args.size >= 2) {
                val target = JukeboxServer.getInstance().getPlayer(args[0])
                if (target == null) {
                    commandSender.sendMessage("§cThe player " + args[0] + " could not be found")
                    return
                }
                val enchantmentValue = args[1].lowercase(Locale.getDefault())
                if (!EnumUtils.isValidEnum(EnchantmentType::class.java, enchantmentValue.uppercase())) {
                    commandSender.sendMessage("§cEnchantment not found.")
                    return
                }
                val enchantmentType = EnchantmentType.valueOf(enchantmentValue.uppercase())
                var level = 1
                if (args.size >= 3) {
                    if (!NumberUtils.isCreatable(args[2])) {
                        commandSender.sendMessage("§cThe level must be a number.")
                        return
                    }
                    level = args[2].toInt()
                }
                val itemInHand = target.getInventory().getItemInHand()
                if (itemInHand.getType() == ItemType.AIR) {
                    commandSender.sendMessage("§cAir can not be enchanted.")
                    return
                }
                val enchantment = itemInHand.getEnchantment(enchantmentType)
                if (enchantment != null) {
                    commandSender.sendMessage("§cThe item already has the enchantment")
                    return
                }
                itemInHand.addEnchantment(enchantmentType, level)
                target.getInventory().setItemInHand(itemInHand)
                commandSender.sendMessage("Enchanting succeeded for " + target.getName())
            }
        } else {
            commandSender.sendMessage("§cYou must be a player.")
        }
    }
}